cromwell_tools/_cromwell_api.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime, timedelta
+import time
+
+import requests
+import requests_mock
+
+from tenacity import retry, wait_exponential, stop_after_delay
+from ._utilities import validate_cromwell_label, prepare_workflow_manifest
+
+
+class WorkflowFailedException(Exception):
+    pass
+
+
+class WorkFlowUnknownException(Exception):
+    pass
+
+
+# todo functools partial for get, post (set the authenticate commands)
+class CromwellAPI:
+
+    # todo remove this if it is not critical for mocking
+    requests = requests
+
+    _status_endpoint = '/api/workflows/v1/{uuid}/status'
```

Comment:
Why define these at the top of the class, instead of in the methods where they are being used? I think the latter would be more readable.

Reply:
Because the old test code was redefining the string literals, and I wanted them to be accessible so they would never be redefined. If the new tests don't need these, then we can move them, but I suspect the mocks might need them.